src/components/PartialSync.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import { Component } from 'react';
+import { db } from '../utils/dbManager';
+import { connect } from "react-redux";
+import { setHistoryId } from "../actions/index";
+import store from "../store/index"
+
+function mapDispatchToProps(dispatch) {
+    return {
+      setHistoryId: historyId => dispatch(setHistoryId(historyId)),
+    };
+  }
+
+
+class PartialSync extends Component {
+    componentDidMount() {
+        this.interval = setInterval(() => this.setState({ time: Date.now() }), 1000);
+    }
+    
+    componentWillUnmount() {
+        clearInterval(this.interval);
+    }
+
+    syncMessageLabels = () => {
+        const historyId = store.getState().historyId;
+        var data = `startHistoryId=${historyId}&historyTypes=labelRemoved&historyTypes=labelAdded`;
+        var xhr = new XMLHttpRequest()
+        xhr.open('GET', `https://gmail.googleapis.com/gmail/v1/users/me/history/?${data}`);
+        xhr.setRequestHeader('Authorization', 'Bearer ' + store.getState().accessToken);
+        xhr.onreadystatechange = () => {
+            if (xhr.readyState !== 4) {
+                return;
+            }
+            if (xhr.status === 200) {
+                const responseJson = JSON.parse(xhr.responseText);
+                const responseHistoryJson = responseJson.history;
+                if(!responseHistoryJson) return;
+                responseHistoryJson.forEach(thread => {
+                    var threadId = null;
+                    var unionLabels = [];
+                    const labelsRemovedJson = thread.labelsRemoved;
+                    if(labelsRemovedJson){
```

Comment:
Lets move all of these into small independent functions. Like handleLabelRemoved, handleLabelsAdded